forc-pkg/src/pkg.rs

```diff
@@ -2011,16 +2014,55 @@ impl PkgTestEntry {
             bail!("Invalid test argument(s) for test: {test_name}.")
         }?;
 
+        /// Given an AST node, checks:
+        /// - it is function,
+        /// - it is a log,
+        ///
+        /// If condition holds, returns type of log's param.
+        fn collect_log_type_from_decl(
+            expr: &TyExpression,
+            log_param_types: &mut Vec<(SourceId, Option<Arc<TypeInfo>>)>,
+            engines: &Engines,
+        ) {
+            if let TyExpressionVariant::FunctionApplication {
+                call_path,
+                arguments,
+                ..
+            } = &expr.expression
+            {
+                let suffix = call_path.suffix.to_string();
+                if suffix == "log" {
+                    let arg_type = arguments
+                        .iter()
+                        .map(|(_, arg_expr)| arg_expr.return_type)
+                        .map(|type_id| engines.te().get(type_id))
+                        .next();
+                    log_param_types.push((expr.span.source_id().cloned().unwrap(), arg_type));
+                }
+            }
+        }
+
+        let mut log_params = Vec::new();
+        for ast_node in test_function_decl.body.contents.iter() {
+            if let TyAstNodeContent::Expression(expr) = &ast_node.content {
+                // We found an expression node, try to extract the arguments if this is an assert.
+                collect_log_type_from_decl(expr, &mut log_params, engines);
```

Comment:
nit: imho, it would be easier to read if the contents of this function were inline here, unless it's going to be called from somewhere else. Having the helper function right above confused me a little at first because it looked like it was calling itself recursively.

Reply:
Yeah, it was recursive at first i designed it, thinking that log can accept multiple params. It looks like it cant i will inline it